apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2ColumnsResourceImpl.java

```diff
@@ -10,130 +11,133 @@
   @Override
-  public Response getAllColumns(String keyspaceName, String tableName, boolean raw) {
+  public Uni<RestResponse<Object>> getAllColumns(
+      String keyspaceName, String tableName, boolean raw) {
     requireNonEmptyKeyspaceAndTable(keyspaceName, tableName);
-    Schema.CqlTable tableDef = getTable(keyspaceName, tableName);
-    List<Sgv2ColumnDefinition> columns = table2columns(tableDef);
-    final Object payload = raw ? columns : new Sgv2RESTResponse<>(columns);
-    return Response.status(Response.Status.OK).entity(payload).build();
+
+    return getTableAsyncCheckExistence(keyspaceName, tableName, true, Response.Status.BAD_REQUEST)
+        .map(t -> table2columns(t))
+        // map to wrapper if needed
+        .map(t -> raw ? t : new Sgv2RESTResponse<>(t))
+        .map(result -> RestResponse.ok(result));
   }
 
   @Override
-  public Response createColumn(
+  public Uni<RestResponse<Object>> createColumn(
       String keyspaceName, String tableName, Sgv2ColumnDefinition columnDefinition) {
     requireNonEmptyKeyspaceAndTable(keyspaceName, tableName);
     final String columnName = columnDefinition.getName();
     if (isStringEmpty(tableName)) {
       throw new WebApplicationException("columnName must be provided", Response.Status.BAD_REQUEST);
     }
-    queryWithTable(
-        keyspaceName,
-        tableName,
-        (tableDef) -> {
-          Column.Kind kind =
-              columnDefinition.getIsStatic() ? Column.Kind.STATIC : Column.Kind.REGULAR;
-          Column columnDef =
-              ImmutableColumn.builder()
-                  .name(columnName)
-                  .kind(kind)
-                  .type(columnDefinition.getTypeDefinition())
+    return queryWithTableAsync(
+            keyspaceName,
+            tableName,
+            (tableDef) -> {
+              Column.Kind kind =
+                  columnDefinition.getIsStatic() ? Column.Kind.STATIC : Column.Kind.REGULAR;
+              Column columnDef =
+                  ImmutableColumn.builder()
+                      .name(columnName)
+                      .kind(kind)
+                      .type(columnDefinition.getTypeDefinition())
+                      .build();
+              return new QueryBuilder()
+                  .alter()
+                  .table(keyspaceName, tableName)
+                  .addColumn(columnDef)
+                  .parameters(PARAMETERS_FOR_LOCAL_QUORUM)
                   .build();
-          return new QueryBuilder()
-              .alter()
-              .table(keyspaceName, tableName)
-              .addColumn(columnDef)
-              .parameters(PARAMETERS_FOR_LOCAL_QUORUM)
-              .build();
-        });
-    return Response.status(Response.Status.CREATED)
-        .entity(Collections.singletonMap("name", columnName))
-        .build();
+            })
+        .map(any -> restResponseCreatedWithName(columnName));
```

Comment:
and is fine not to peak at the response at all, guess it's fine..

Reply:
That was the existing behavior, FWTW.